app/src/main/java/com/joocoding/android/app/githubsearch/network/GithubRetrofit.kt

```diff
@@ -0,0 +1,21 @@
+package com.joocoding.android.app.githubsearch.network
+
+import com.joocoding.android.app.githubsearch.service.SearchService
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object GithubRetrofit {
+    private const val BASE_URL = "https://api.github.com/"
+    val githubRetrofit: SearchService
+
+    init {
+        val retrofit: Retrofit = Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+
+        githubRetrofit = retrofit.create(SearchService::class.java)
+    }
```

Comment:
개인적인 취향이라 참고만 해주세요!
koltin에서 생성자에서 변수를 초기화한다면 by lazy를 쓰는 것이 깔끔해보이더라구요

```suggestion
    val githubRetrofit: SearchService by lazy {
        val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        githubRetrofit = retrofit.create(SearchService::class.java)
    }
```